supervisor/jobs/__init__.py

```diff
@@ -282,8 +282,10 @@ def _on_job_change(
                 # reporting shouldn't raise and break the active job
                 continue
 
-            progress = sync.starting_progress + (
-                sync.progress_allocation * job_data["progress"]
+            progress = min(
+                100,
+                sync.starting_progress
+                + (sync.progress_allocation * job_data["progress"]),
```

Comment:
This isn't actually what caught the bug. What caught the bug was attempting to sync to a parent job that was already 100% complete. I considered having it skip the parent sync when the parent was like this but decided since that should never happen and was always a bug it would be better to raise so we can fix it.

If progress goes over 100 here that is also probably a developer error due to an incorrect value for progress allocation which should be fixed. But it could also be a result of float drift which is not a bug so for that reason I decided to prevent it from going over 100 with a `min`.